core/primitives/src/runtime/config_store.rs

```diff
@@ -134,6 +132,26 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_data_roundtrip_is_more_expensive() {
+        // We have an assumption that the deepest receipts we can create is by creating recursive
+        // function call promises (calling function call from a function call).
+        // If the cost of a data receipt is cheaper than the cost of a function call, then it's
+        // possible to create a promise with a dependency which will be executed in two blocks that
+        // is cheaper than just two recursive function calls.
+        // That's why we need to enforce that the cost of the data receipt is not less than a
+        // function call. Otherwise we'd have to modify the way we compute the maximum depth.
```

Comment:
Interesting. With the new change it looks like this test will fail. @Longarithm

Reply:
I took a closer look and I think this test should be removed:
> If the cost of a data receipt is cheaper than the cost of a function call, then it's
> possible to create a promise with a dependency which will be executed in two blocks that
> is cheaper than just two recursive function calls.

If that was **really** the case, recursive function calls would not make sense because we could always create promises with dependencies. And we actually charge function call fees on promise creation here:
 
https://github.com/near/nearcore/blob/a20c12ec02d7f22550139da38838efd1fba75ede/runtime/near-vm-logic/src/logic.rs#L1530